migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/column.rs

```diff
@@ -163,10 +170,10 @@ impl ColumnChanges {
     }
 
     pub(crate) fn only_default_changed(&self) -> bool {
-        matches!(self.changes, [None, None, None, Some(ColumnChange::Default)])
+        matches!(self.changes, [None, None, None, Some(ColumnChange::Default), None])
     }
 
     pub(crate) fn column_was_renamed(&self) -> bool {
-        matches!(self.changes, [Some(ColumnChange::Renaming), _, _, _])
+        matches!(self.changes, [Some(ColumnChange::Renaming), _, _, _, _])
```

Comment:
😄

Reply:
Yes it's not ideal, I'm still looking for a better data structure for the job. Basically a bitmask would work, but I need a way to force users of the API to handle every possible type of change exhaustively (there will be more in the future). I tried a trait-based approach but it didn't feel right.

Reply:
(and the array part is for avoiding an allocation, premature optimization and all that)

Reply:
a HashSet would work too (but with allocation)

Reply:
Do you think allocation is necessary a problem here?

Reply:
Not really. It still feels bad though. We will generate a `ColumnChanges` object for every single field in the schema at least 4 times during a normal `prisma migrate save` (once for infer, once for apply, multiply by two because we also generate down migrations).